client-flink/flink-1.20/src/main/java/org/apache/celeborn/plugin/flink/tiered/CelebornChannelBufferManager.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.celeborn.plugin.flink.tiered;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+import static org.apache.flink.util.Preconditions.checkState;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+import org.apache.flink.core.memory.MemorySegment;
+import org.apache.flink.runtime.io.network.buffer.Buffer;
+import org.apache.flink.runtime.io.network.buffer.BufferListener;
+import org.apache.flink.runtime.io.network.buffer.BufferPool;
+import org.apache.flink.runtime.io.network.buffer.BufferRecycler;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage.TieredStorageMemoryManager;
+import org.apache.flink.util.ExceptionUtils;
+
+public class CelebornChannelBufferManager implements BufferListener, BufferRecycler {
+
+  /** The queue to hold the available buffer when the reader is waiting for buffers. */
+  private final Queue<Buffer> bufferQueue;
+
+  private final TieredStorageMemoryManager memoryManager;
+
+  private final CelebornChannelBufferReader bufferReader;
+
+  /** The tag indicates whether it is waiting for buffers from the buffer pool. */
+  private boolean isWaitingForFloatingBuffers;
+
+  /** The total number of required buffers for the respective input channel. */
+  private int numRequiredBuffers = 0;
+
+  public CelebornChannelBufferManager(
+      TieredStorageMemoryManager memoryManager, CelebornChannelBufferReader bufferReader) {
+    this.memoryManager = checkNotNull(memoryManager);
+    this.bufferReader = checkNotNull(bufferReader);
+    this.bufferQueue = new LinkedList<>();
+  }
+
+  @Override
+  public boolean notifyBufferAvailable(Buffer buffer) {
+    if (bufferReader.isClosed()) {
+      return false;
+    }
+    int numBuffers = 0;
+    boolean isBufferUsed = false;
+    try {
+      synchronized (bufferQueue) {
+        checkState(
```

Comment:
If `isWaitingForFloatingBuffers` is false, should this return false instead of checking state?